### PR TITLE
Async grade book downloading

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
### Background
Issue https://github.com/mitocw/edx-platform/issues/108#issuecomment-161961956
In this PR i copied grade book downloading mechanism from ``Instructor dashboard/ Data download``

### What is done in this PR
**Studio Updates:** None.
**LMS Updates:** 
 - CCX coach can request for download grade book.
 - Once grade book is download it will be available in list of ready to download grade books.
 - From there user can download grade book. 

@pdpinch @giocalitri 

screen shot 2015-12-09 at 5 24 00 pm

### CODE REVIEW

File: lms/djangoapps/ccx/urls.py [hunk 1]
```diff
@@ -27,6 +27,10 @@
 
     url(r'^ccx_grades.csv$',
         'ccx.views.ccx_grades_csv', name='ccx_grades_csv'),
+
+    url(r'^list_ready_downloads$',
```
Comment (pdpinch) [inline lms/djangoapps/ccx/urls.py]: Is this the same (relative) path as the one used in the instructor dashboard?

File: lms/djangoapps/ccx/views.py [hunk 2]
```diff
@@ -612,6 +624,28 @@ def ccx_gradebook(request, course, ccx=None):
         })
 
 
+@transaction.non_atomic_requests
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@coach_dashboard
+def list_ready_downloads(request, course, ccx=None):
```
Comment (pdpinch) [inline lms/djangoapps/ccx/views.py]: Insted of copying & pasting, is there any way to re-use the code from the instructor dashboard?

**Comment by pdpinch** [comment]
In general, it seems like a lot of generic code has specific references to gradebook in it. Can we remove that from the function names, unless they deal specifically with the gradebook?

For example, https://github.com/mitocw/edx-platform/pull/157/files#diff-15eeaf5424706199857a4046fdac02daR25

**Comment by pdpinch** [comment]
Despite my comments, I think this is OK to push up to edX.

**Comment by amir-qayyum-khan** [comment]
ok, i will push it tomorrow morning.

**Comment by amir-qayyum-khan** [comment]
@pdpinch done with incorporating ur feedback, made upstream PR https://github.com/edx/edx-platform/pull/11092

**Pull request closed without merging**